### PR TITLE
Add scaling rules to resource pool spec

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: api/resourcepool/v1/resourcepool_types.go [hunk 1]
```diff
@@ -33,7 +33,24 @@ type ResourceDemandStatus struct {
 	StatusMessage string `json:"statusMessage"`
 
 	// Time at which the report was made.
-	ReportedAt uint64 `json:"reportedAt"`
+	ReportedAt int64 `json:"reportedAt"`
```
Comment (joshi-keyur) [inline api/resourcepool/v1/resourcepool_types.go:36]: Curious why we are going from unsigned to signed for timestamp.

File: api/resourcepool/v1/resourcepool_types.go [hunk 2]
```diff
@@ -33,7 +33,24 @@ type ResourceDemandStatus struct {
...
+	MaxIdle int64 `json:"maxIdle"`
+
+	// Resource pool minimum size limit (minimum number of shapes that can be ever allocated).
+	MinSize int64 `json:"minSize"`
+
+	// Resource pool maximum size limit (maximum number of shapes that can be ever allocated).
+	MaxSize int64 `json:"maxSize"`
+
+	// Set to true to enable auto scaling of this resource pool.
+	AutoScaleEnabled bool `json:"AutoScaleEnabled"`
```
Comment (joshi-keyur) [inline api/resourcepool/v1/resourcepool_types.go:53]: `s/AutoScaleEnabled/autoScaleEnabled/`?

File: api/resourcepool/v1/resourcepool_types.go (hunk 1 above)
Comment (tbak) [reply api/resourcepool/v1/resourcepool_types.go:36]: We use `int64` type everywhere. This is one outlier.

File: api/resourcepool/v1/resourcepool_types.go [hunk 3]
```diff
@@ -33,7 +33,24 @@ type ResourceDemandStatus struct {
 	StatusMessage string `json:"statusMessage"`
 
 	// Time at which the report was made.
-	ReportedAt uint64 `json:"reportedAt"`
+	ReportedAt int64 `json:"reportedAt"`
+}
+
+type ResourcePoolScalingRules struct {
```
Comment (joshi-keyur) [inline api/resourcepool/v1/resourcepool_types.go:39]: I am curious when we name type as plural.

Comment (tbak) [reply api/resourcepool/v1/resourcepool_types.go:39]: Because we have several rules in this data structure. I am ok with using singular form.

**Review by corindwyer — APPROVED** [review]

**Review by fabiokung — APPROVED** [review]

**Review by joshi-keyur — APPROVED** [review]

**Pull request merged**